src/SourceBrowser/src/SourceIndexServer/Helpers.cs

```diff
@@ -3,93 +3,27 @@
 using System.Net.Http;
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Microsoft.SourceBrowser.SourceIndexServer.Models;
 
 namespace Microsoft.SourceBrowser.SourceIndexServer
 {
     public static class Helpers
     {
-        public static async Task ProxyRequestAsync(this HttpContext context, HttpClient client, string targetUrl, Action<HttpRequestMessage> configureRequest = null)
+        public static async Task ProxyRequestAsync(this HttpContext context, string targetPath, Action<HttpRequestMessage> configureRequest = null)
         {
-            using (var req = new HttpRequestMessage(HttpMethod.Get, targetUrl))
+            var fs = new AzureBlobFileSystem(IndexProxyUrl);
+            context.Response.Headers.Append("Content-Type", "text/html");
+            using (var data = fs.OpenSequentialReadStream(targetPath))
             {
-                foreach (var (key, values) in context.Request.Headers)
-                {
-                    switch (key.ToLower())
-                    {
-                        // We shouldn't copy any of these request headers
-                        case "host":
-                        case "authorization":
-                        case "cookie":
-                        case "content-length":
-                        case "content-type":
-                            continue;
-                        default:
-                            req.Headers.TryAddWithoutValidation(key, values.ToArray());
-                            break;
-                    }
-                }
-
-                configureRequest?.Invoke(req);
-
-                HttpResponseMessage res = await client.SendAsync(req, HttpCompletionOption.ResponseHeadersRead).ConfigureAwait(false);
-                context.Response.RegisterForDispose(res);
-
-                foreach (var (key, values) in res.Headers)
-                {
-                    switch (key.ToLower())
-                    {
-                        // Remove headers that the response doesn't need
-                        case "set-cookie":
-                        case "x-powered-by":
-                        case "x-aspnet-version":
-                        case "server":
-                        case "transfer-encoding":
-                        case "access-control-expose-headers":
-                        case "access-control-allow-origin":
-                            continue;
-                        default:
-                            if (!context.Response.Headers.ContainsKey(key))
-                            {
-                                context.Response.Headers[key] = values.ToArray();

```

Comment:
Did this copy over any headers here, or below that would be interesting to set?  Probably can examine in F12 tools here as well.

Reply:
Good question.

This is the current set of headers in this PR:

<img width="334" alt="image" src="https://github.com/dotnet/source-indexer/assets/308992/e7131a72-68fd-41d4-b7d6-4cbaf5ef6d06">

And here's the old set of headers from the site in prod:

<img width="784" alt="image" src="https://github.com/dotnet/source-indexer/assets/308992/b9457fdf-cdcb-49c5-9010-57f7bfd8624d">

I didn't consider any of the missing entries essential, except for Content-Type which is needed for anything at all to render.

Reply:
yes and `Etag`

Reply:
Also `Permissions-Policy` might be something we need?

Reply:
It looks like `Etag` and `Content-Md5` are properties exposed on the raw blob, so I am 99% I can yoink those programatically & inject them. I don't know about `Permissions-Policy`

<img width="350" alt="image" src="https://github.com/dotnet/source-indexer/assets/308992/8b14f070-eed6-418b-bccd-fd87daba993d">

Reply:
I'm testing a few more headers, including passing through `Content-Type` instead of hardcoding. However I don't know how to represent the `Permissions-Policy` header based on the parent `BlobProperties` object

Reply:
Looking at the docs for `Permissions-Policy` we might not need it, seems mostly for iframes